src/main/java/com/mathworks/ci/RunMatlabTestsStep.java

```diff
@@ -155,9 +195,11 @@ private String getInputArgs() {
         inputArgs.add("'Test'");
 
         args.forEach((key, val) -> {
-            if(key.equals("SourceFolder") || key.equals("SelectByFolder") && val != null){
+            if (key.equals("SourceFolder") || key.equals("SelectByFolder")
+                    || key.equals("UseParallel")
+                    || key.equals("Strict") && val != null) {
```

Comment:
Is this intended to work as written or as
```java
            if ((key.equals("SourceFolder") || key.equals("SelectByFolder")
                    || key.equals("UseParallel")
                    || key.equals("Strict")) && val != null) {
```
I think `&&` gets evaluated before `||` in java so having `val == nul` might only compare with the `key.equals("Strict")` part of this expression.

Reply:
If you see the same condition being evaluated for more than one branch of an if/else, it usually needs a refactor. In this case, you could pull out the `val != null` as a separate block.

```java
if (val == null) {
    continue;
}
if (key...) {
} else {
}
```
OR
```java
if (val != null) {
    if (key...) {
    } else {
    }
}
```

Reply:
Done.